### PR TITLE
Remove installer-e2e-aws-csi jobs

### PR DESCRIPTION
**Pull request by jsafrane**
These jobs do not have corresponding entries in config/openshift/release and we've been dragging them since OCP 4.5. [They do not work any longer](https://testgrid.k8s.io/redhat-openshift-ocp-release-4.14-informing#release-openshift-ocp-installer-e2e-aws-csi-4.14).

There are better / more maintained periodic jobs for AWS CSI driver in each branch like [periodic-ci-openshift-release-master-nightly-4.15-e2e-aws-csi](https://github.com/openshift/release/blob/267fb722a8b21b3b2cef51ef75534c1371a4273c/ci-operator/config/openshift/release/openshift-release-master__nightly-4.15.yaml#L648) and [they are reasonably green](https://testgrid.k8s.io/redhat-openshift-ocp-release-4.14-informing#periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-csi).

Since these were the last jobs that used `installer-custom-test-image` template, I am removing the template too as instructed by `make template-allowlist`.

### CODE REVIEW

File: ci-operator/jobs/openshift/release/openshift-release-release-4.10-periodics.yaml [hunk 1]
```diff
@@ -1205,108 +1205,3 @@ periodics:
...
       secret:
         secretName: manifest-tool-local-pusher
-- agent: kubernetes
-  cluster: build01
-  decorate: true
-  interval: 24h
-  labels:
-    job-release: "4.10"
-    pj-rehearse.openshift.io/can-be-rehearsed: "true"
-  name: release-openshift-ocp-installer-e2e-aws-csi-4.10
```
Comment (bertinatto) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.10-periodics.yaml:1215]: I see references to these jobs in `core-services/template-deprecation/_allowlist.yaml`: https://github.com/openshift/release/blob/523b9fde129be2d6da5bf0da5ce160b17f27973b/core-services/template-deprecation/_allowlist.yaml#L79

**Review by bertinatto — COMMENTED** [review]
`make template-allowlist` does't update `core-services/template-deprecation/_allowlist.yaml`.
/lgtm

File: ci-operator/jobs/openshift/release/openshift-release-release-4.10-periodics.yaml (hunk 1 above)
Comment (jsafrane) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.10-periodics.yaml:1215]: I removed it from the _allowlist.yaml in a new commit. `make template-allowlist` then asked for removal of the template, so I removed it.

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Pull request merged**